src/Servers/Kestrel/Kestrel/test/KestrelConfigurationLoaderTests.cs

```diff
@@ -1224,6 +1226,124 @@ public void Reload_RerunsNamedEndpointConfigurationOnChange()
         Assert.Equal(1, foundUnchangedCount);
     }
 
+    [Fact]
+    public void MultipleLoads_ConfigureBetween()
+    {
+        var currentConfig = new ConfigurationBuilder().AddInMemoryCollection(new[]
+        {
+            new KeyValuePair<string, string>("Endpoints:A:Url", "http://*:5000"),
+            new KeyValuePair<string, string>("Endpoints:B:Url", "http://*:5001"),
+        }).Build();
+
+        var mockConfig = new Mock<IConfiguration>();
+        mockConfig.Setup(c => c.GetSection(It.IsAny<string>())).Returns<string>(currentConfig.GetSection);
+        mockConfig.Setup(c => c.GetChildren()).Returns(currentConfig.GetChildren);
+
+        var serverOptions = CreateServerOptions();
+        serverOptions.Configure(mockConfig.Object, reloadOnChange: false);
+        var oldConfigurationLoader = serverOptions.ConfigurationLoader;
+        
+        serverOptions.ConfigurationLoader.Load();
+
+        mockConfig.Verify(c => c.GetSection(It.IsAny<string>()), Times.AtLeastOnce);
+
+        mockConfig.Invocations.Clear();
+
+        serverOptions.Configure(mockConfig.Object, reloadOnChange: false);
+        var newConfigurationLoader = serverOptions.ConfigurationLoader;
+        Assert.NotSame(oldConfigurationLoader, newConfigurationLoader);
+
+        serverOptions.ConfigurationLoader.Load();
+
+        // The change token is stored on the loader, so replacing the loader replaces the token.
+        mockConfig.Verify(c => c.GetSection(It.IsAny<string>()), Times.AtLeastOnce);
+    }
+
+    [Theory]
+    [InlineData(true, true)]
+    [InlineData(false, true)]
+    [InlineData(true, false)]
+    [InlineData(false, false)]
+    public void MultipleLoads_ConfigurationChanges(bool changeBeforeInitialLoad, bool changeAfterInitialLoad)
+    {
+        var currentConfig = new ConfigurationBuilder().AddInMemoryCollection(new[]
+        {
+            new KeyValuePair<string, string>("Endpoints:A:Url", "http://*:5000"),
+            new KeyValuePair<string, string>("Endpoints:B:Url", "http://*:5001"),
+        }).Build();
+
+        var mockReloadToken = new Mock<IChangeToken>();
+        mockReloadToken.SetupGet(t => t.HasChanged).Returns(changeBeforeInitialLoad);
```

Comment:
Why are we parameterizing this return value when below we assert `mockReloadToken.VerifyGet(t => t.HasChanged, Times.Never);` right before reconfiguring it to return `changeAfterInitialLoad`? It couldn't possibly make a difference given we verify it's never called.

Reply:
I agree that it has no effect on the behavior - I thought it was easier to reason about with a more complete implementation.

Reply:
I think it makes things more confusing because it makes you wonder why the change token is even being checked that early. And it ends up adding two redundant test cases.

Reply:
It's true that the tests are redundant with the current product code.  My goal in writing the test was to warn people when they make subtly incorrect refactorings.  If the goal is only to test the code as it exists today, then I can get rid of the check.

Reply:
Removed.